scripts/mixins/job_special.lua

```diff
@@ -0,0 +1,52 @@
+-- for mobs with job-specific special ability
+
+require("scripts/globals/mixins");
+require("scripts/globals/status");
+
+g_mixins = g_mixins or {};
+
+g_mixins.job_special = function(mob)
+
+    mob:addListener("SPAWN", "JOB_SPECIAL_SPAWN", function(mob)
+        local defaultAbility = {
+            [JOBS.WAR] = jobSpec.MIGHTY_STRIKES,
+            [JOBS.MNK] = jobSpec.HUNDRED_FISTS,
+            [JOBS.WHM] = jobSpec.BENEDICTION,
+            [JOBS.BLM] = jobSpec.MANAFONT,
+            [JOBS.RDM] = jobSpec.CHAINSPELL,
+            [JOBS.THF] = jobSpec.PERFECT_DODGE,
+            [JOBS.PLD] = jobSpec.INVINCIBLE,
+            [JOBS.DRK] = jobSpec.BLOOD_WEAPON,
+            [JOBS.BST] = jobSpec.FAMILIAR,
+            [JOBS.BRD] = jobSpec.SOUL_VOICE,
+            [JOBS.SAM] = jobSpec.MEIKYO_SHISUI,
+            [JOBS.NIN] = jobSpec.MIJIN_GAKURE,
+            [JOBS.DRG] = jobSpec.CALL_WYVERN,
+            [JOBS.SMN] = jobSpec.ASTRAL_FLOW,
+        };
+
+        mob:setLocalVar("usedJobSpec", 0);
+        mob:setLocalVar("useJobSpecAtPct", math.random(40,60));
+        local ability = defaultAbility[mob:getMainJob()];
+        if (mob:getLocalVar("jobSpec") == 0 and ability ~= nil) then
+            mob:setLocalVar("jobSpec", ability);
+        end
+    end);
+    
+    mob:addListener("WEAPONSKILL_STATE_ENTER", "JOB_SPECIAL_WS", function(mob, skillID)
```

Comment:
status_effect_tick is probably a better event to listen on (ticks every 3 sec), unless you know that they only 2hour right after using a mob ability

Reply:
thanks, was trying to figure that out.

Reply:
they should all (hopefully) be listed in documentation/AIEvents.txt

Reply:
Shelbyz showed how to add combat_tick and roam_tick, so I've implemented that here.